### ISSUE TITLE
v2 breaks the way --ginkgo.parallel.process -and -ginkgo.parallel.total used to work

### ISSUE THREAD

**Issue by tomastigera** [onsi/ginkgo#971]
It was possible to run independent batches on different machines that would just execute their respective shard with v1. v2 can spin up multiple processes in parallel that require `--parallel.host` to be set (by default by CLI) so that a server would coordinate the shards. Is there either a way how to spin up the server independently or to turn this behaviour off? Parallelization within a single machine only is very limiting.

Also --dry-run does not work in parallel mode anymore, therefore it is not possible to print out which tests would be executed by the batch. But that seems to be just result of the above.

**Comment by onsi** [comment]
Hey @tomastigera 

Yes 2.0 removed support for that kind of sharding (which itself was a holdover of how early versions of Ginkgo used to distribute specs in parallel).  I'd always designed Ginkgo with the ability to run processes on remote workers in mind but had never seen anyone actually put that support to use.  So it's cool to hear you're interested.

I'd be happy to reintroduce it - but not via sharding.  I'd rather make it easy to spin up a server and allow you to launch test suites that connect with the central server.  This will allow for report aggregation and additional coordination between the nodes.

Would you rather import a library and write your own code to spin up a server?  Or would you prefer to have a cli command that starts up a server and waits for spec runners to connect?

**Comment by tomastigera** [comment]
Hey @onsi  yeah I can see it is all ready for multi-worker setup :+1: 

> Would you rather import a library and write your own code to spin up a server? Or would you prefer to have a cli command that starts up a server and waits for spec runners to connect?

Depends on what the server needs to know. I compile the test into a single binary that is distributed across the workers. Perhaps what I would like to have is that the same binary san act both as a client where I would add `--parallel.host` and a server which I would run like `--server <port>` ?

When there is a failure each node keeps quite a verbose output for diagnostics. Not sure if it is practical to ship it to the server, but it would be nice to have and option to keep it local or ship it.

**Comment by fasaxc** [comment]
@onsi FYI we're running in a cloud CI so we can't _just_ start VMs, they're workers started by the CI system with particular parameters.  The CI provider lets us start up n parallel workers, each with its own shard number passed in by env var, but there's no ability to start up n workers, with a central coordination server and connectivity between them.
